bentoml/_internal/io_descriptors/multipart.py

```diff
@@ -83,7 +144,35 @@ async def from_http_request(self, request: Request) -> MultipartIO:
             res[k] = v
         return res
 
-    async def to_http_response(self, obj: MultipartIO) -> Response:
+    @t.overload
+    async def to_http_response(self, obj: t.Dict[str, str]) -> Response:
+        ...
+
+    @t.overload
+    async def to_http_response(self, obj: t.Dict[str, "JSONType"]) -> Response:
+        ...
+
+    @t.overload
+    async def to_http_response(self, obj: t.Dict[str, "ImageType"]) -> Response:
+        ...
+
+    @t.overload
+    async def to_http_response(
+        self, obj: t.Dict[str, "np.ndarray[t.Any, np.dtype[t.Any]]"]
+    ) -> Response:
+        ...
+
+    @t.overload
+    async def to_http_response(self, obj: t.Dict[str, "FileLike"]) -> Response:
+        ...
+
+    @t.overload
```

Comment:
we should have just Union here rather than overloads?

Reply:
for some reason pyright make a false negative with Union type